pkg/user/tx_client.go

```diff
@@ -379,40 +393,175 @@ func (client *TxClient) BroadcastTx(ctx context.Context, msgs []sdktypes.Msg, op
 }
 
 func (client *TxClient) broadcastTx(ctx context.Context, txBytes []byte, signer string) (*sdktypes.TxResponse, error) {
-	txClient := sdktx.NewServiceClient(client.grpc)
-	resp, err := txClient.BroadcastTx(
-		ctx,
-		&sdktx.BroadcastTxRequest{
-			Mode:    sdktx.BroadcastMode_BROADCAST_MODE_SYNC,
-			TxBytes: txBytes,
-		},
+	numConns := len(client.conns)
+	if numConns == 0 {
+		return nil, errors.New("no connections available in TxClient")
```

Comment:
this check should not exist here - it should exist in constructor. `TxClient` should fail to build if there is not at least 1 grpc connection provided to it.